src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -1390,9 +1387,9 @@ public enum DarknessAblaze implements LogicCardInfo {
         bwAbility "Primal Law", {
           text "If this Pokémon is your Active Pokémon, your opponent can’t play any Pokémon from their hand to evolve their Pokémon."
           delayedA {
-            before PLAY_EVOLUTION, {
-              if (self.active && bg.currentTurn == self.owner.opposite && ef.cardToPlay.player.pbg.hand.contains(ef.cardToPlay)) {
-                wcu "$self.name's Primal Law prevents playing Pokémon from your hand to evolve your Pokémon"
+            before EVOLVE_STANDARD, {
+              if (self.active && bg.currentTurn == self.owner.opposite && (ef as EvolveStandard).evolutionCard.player.pbg.hand.contains(ef.evolutionCard)) {
+                wcu "$self's Primal Law prevents playing Pokémon from your hand to evolve your Pokémon"
```

Comment:
Really minor detail, but it seems two warning conflict with each other when directly playing an evolution from hand.

Old Rare Candy seems to skip this prevention still, but that's not really important (and shouldn't even be a problem for Unlimited, whenever common code is implemented for it like with Pokemon Catcher)

Ditto Prism Star seems to bypass this effect. Same with Ascension DNA in Eevee-GX, and Quick-Ripening Herb in Meganium (Lost Thunder).

Reply:
> Really minor detail, but it seems two warning conflict with each other when directly playing an evolution from hand

Oh, forgot to pass a screenshot I took of that:

![image](https://user-images.githubusercontent.com/1235066/89599928-2abc3600-d837-11ea-9144-be43ab66dcad.png)

Reply:
The stacking issue seems to be something Vaadin 14 fixes https://github.com/vaadin/framework/issues/676#issuecomment-544166861

Playing an evolution directly from hand shouldn't cause stacked notifications now, however Rare Candy does. A lot. 4 of them. 2 of which are empty currently due to a bug in the main repo.

Reply:
> however Rare Candy does
Doesn't seem to pop em up here now, lol.

Hey, if it works it works 😛